src/analysis_file.py

```diff
@@ -0,0 +1,238 @@
+            for code in plan.code_scheme.codes:
+                show_matrix_keys.append(f"{plan.analysis_file_key}{code.string_value}")
+
+            AnalysisKeys.set_matrix_keys(
+                user, data, show_matrix_keys, plan.code_scheme, plan.coded_field, plan.analysis_file_key)
+
+            matrix_keys.extend(show_matrix_keys)
+
+        binary_keys = [plan.binary_analysis_file_key
+                       for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS
+                       if plan.binary_analysis_file_key is not None]
+
+        equal_keys = ["uid", "operator"]
+        equal_keys.extend(demog_keys)
+        concat_keys = [plan.raw_field for plan in PipelineConfiguration.RQA_CODING_PLANS]
+        follow_up_concat_keys = [plan.raw_field for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS]
+        concat_keys.extend(follow_up_concat_keys)
+
+        bool_keys = [
+            consent_withdrawn_key,
+            "drama_promo",
+            "magazine_promo",
+            "radio_drama",
+            "non_logical_time",
+            "radio_drama_participation_s01e01",
+            "radio_drama_participation_s01e02",
+            "radio_drama_participation_s01e03",
+            "radio_drama_participation_s01e04",
+            "radio_drama_participation_s01e05",
+            "radio_drama_participation_s01e06",
+            "radio_drama_participation_s01e07",
+            "radio_magazine_participation_s01e01",
+            "radio_magazine_participation_s01e02",
+        ]
+
+        # Export to CSV
+        export_keys = ["uid", "operator"]
+        export_keys.extend(bool_keys)
+        export_keys.extend(matrix_keys)
+        export_keys.extend(binary_keys)
+        export_keys.extend(concat_keys)
+        export_keys.extend(demog_keys)
+
+        # Set consent withdrawn based on presence of data coded as "stop"
+        ConsentUtils.determine_consent_withdrawn(
+            user, data, PipelineConfiguration.DEMOG_CODING_PLANS, consent_withdrawn_key)
+
+        # Set consent withdrawn based on stop codes from radio question and follow up survey codes answers
+        for td in data:
+            for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+                if td[f"{plan.analysis_file_key}{Codes.STOP}"] == Codes.MATRIX_1:
+                    td.append_data({consent_withdrawn_key: Codes.TRUE},
+                                   Metadata(user, Metadata.get_call_location(), time.time()))
+
+                if plan.binary_code_scheme is not None:
+                    if td[plan.binary_coded_field]["CodeID"] == \
+                            plan.binary_code_scheme.get_code_with_control_code(Codes.STOP).code_id:
+                        td.append_data({consent_withdrawn_key: Codes.TRUE},
+                                       Metadata(user, Metadata.get_call_location(), time.time()))
+
+        '''
```

Comment:
If it's all working I think we can remove this now

(if we wanted this level of helpful debugging information in future, Core is the best place for it, ideally in a redesigned FoldTracedData routine)

Reply:
Removed thanks!